test-programs/sdk-test-program/tests/test.rs

```diff
@@ -0,0 +1,306 @@
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, Signer};
+use solana_sdk::transaction::{Transaction, TransactionError};
+
+#[tokio::test]
+async fn test_sdk_test() {
+    let (mut rpc, env) =
+        setup_test_programs_with_accounts_v2(Some(vec![(String::from("sdk_test"), sdk_test::ID)]))
+            .await;
+    let payer = rpc.get_payer().insecure_clone();
+
+    let mut test_indexer: TestIndexer<ProgramTestRpcConnection> = TestIndexer::new(
+        &[StateMerkleTreeAccounts {
+            merkle_tree: env.merkle_tree_pubkey,
+            nullifier_queue: env.nullifier_queue_pubkey,
+            cpi_context: env.cpi_context_account_pubkey,
+        }],
+        &[AddressMerkleTreeAccounts {
+            merkle_tree: env.address_merkle_tree_pubkey,
+            queue: env.address_merkle_tree_queue_pubkey,
+        }],
+        true,
+        true,
+    )
+    .await;
+
+    let mut remaining_accounts = RemainingAccounts::default();
+
+    let merkle_context = MerkleContext {
+        merkle_tree_pubkey: env.merkle_tree_pubkey,
+        nullifier_queue_pubkey: env.nullifier_queue_pubkey,
+        leaf_index: 0,
+        queue_index: None,
+    };
+    let merkle_context = pack_merkle_context(merkle_context, &mut remaining_accounts);
+
+    let address_merkle_context = AddressMerkleContext {
+        address_merkle_tree_pubkey: env.address_merkle_tree_pubkey,
+        address_queue_pubkey: env.address_merkle_tree_queue_pubkey,
+    };
+
+    let address_seed = derive_address_seed(&[b"sdk_test"], &sdk_test::ID);
+    let address = derive_address(&address_seed, &address_merkle_context);
+
+    let address_merkle_context =
+        pack_address_merkle_context(address_merkle_context, &mut remaining_accounts);
+
+    let account_compression_authority = get_cpi_authority_pda(&PROGRAM_ID_LIGHT_SYSTEM);
+    let registered_program_pda = Pubkey::find_program_address(
+        &[PROGRAM_ID_LIGHT_SYSTEM.to_bytes().as_slice()],
+        &PROGRAM_ID_ACCOUNT_COMPRESSION,
+    )
+    .0;
+
+    with_nested_data(
+        1, // one
+        &mut rpc,
+        &mut test_indexer,
+        &env,
+        &mut remaining_accounts,
+        &payer,
+        &address,
+        &merkle_context,
+        &address_merkle_context,
+        &account_compression_authority,
+        &registered_program_pda,
+        &PROGRAM_ID_LIGHT_SYSTEM,
+    )
+    .await
+    .unwrap();
+
+    println!("done");
+
+    // Check that it was created correctly.
+    let compressed_accounts = test_indexer.get_compressed_accounts_by_owner(&sdk_test::ID);
+    assert_eq!(compressed_accounts.len(), 1);
+    let compressed_account = &compressed_accounts[0];
+    let record = &compressed_account
+        .compressed_account
+        .data
+        .as_ref()
+        .unwrap()
+        .data;
+    let record = MyCompressedAccount::deserialize(&mut &record[..]).unwrap();
+    assert_eq!(record.nested.one, 1);
+
+    println!("done 2");
+    // update_record(
```

Comment:
I would remove this commented code - it's from a different example (DNS) and I think we can just add an update operation here later. I can take it as a next PR from my side.

Reply:
thanks - i added test coverage now for update_nested_data for completeness